landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+             0.55372743  0.55372743  0.55372743  0.55372743]
+
+
+
+     References
+     ----------
+     Tang, H., McGuire, L. A., Rengers, F. K., Kean, J. W., Staley,
+     D. M., & Smith, J. B. (2019). Developing and Testing Physically
+     Based Triggering Thresholds for Runoff-Generated Debris Flows.
+     Geophysical Research Letters, 46(15), 8830–8839.
+     https://doi.org/10.1029/2019GL083623
+
+    """
+
+    _name = "DimensionlessDischargeModel"
+
+    _unit_agnostic = False
+
+    _info = {
+        "dimensionless_discharge": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "dimensionless_discharge_above_threshold": {
+            "dtype": bool,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "flux": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "flux in each stream segment",
+        },
+        "d50": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "soil grain size average in stream segment",
+        },
+        "dem_values": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "elevation stream segment",
+        },
+    }
+
+    def __init__(self, grid, soil_density=1330, water_density=997.9, C=12.0, N=0.85):
+        """Initialize the DimensionlessDischarge.
+
+        Parameters
+        ----------
+        soil_density : float, optional (defaults to 1330)
+            density of soil (kg/m^3)
+        water_density : float, optional (defaults to 997.9)
+            density of water (kg/m^3)
+        C : float, optional (defaults to 12.0)
+            Numerator of the debris flow threshold equation; Empirically
+            derived constant (See Tang et al. 2019)
+        N : float, optional (defaults to 0.85)
+            Exponent for slope in the denominator of the debris flow
+            threshold equation; Empirically derived constant (See Tang
+            et al. 2019)
+        """
+
+        super().__init__(grid)
+
+        # Store parameters
+        self._soil_density = soil_density
+        self._C = [C] * grid.number_of_nodes
+        self._N = [N] * grid.number_of_nodes
+        # change DEM values into slope of a stream segment
+        self._stream_slopes = grid.calc_slope_at_node(elevs="dem_values")
+        self.water_density = water_density
```

Comment:
This should be a private variable.
```python
self._water_density = water_density
```
If you intend for users to change the density of water after the component has been instantiated, you should set it as a property and add a setter. So something like,
```python
@property
def water_density(self):
    return self._water_density

@water_density.setter
def water_density(self, new_value):
    if water_density > 0.0:
        self._water_density = new_values
    else:
        raise ValueError(f"water density must be greater than zero ({new_value})")

Reply:
Made into a private variable.